### PR TITLE
Remove secrets and fix shell step in task definition redeploy

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description

I'm modifying the composite action here and then will backport the fixes to the template in the infrastructure repository where it is generated once it's fixed.

## Testing Instructions

See if it works on this branch, remove the testing code, then merge and hope it also works on main.

Update: it worked! https://github.com/WordPress/openverse-frontend/runs/6828684979?check_suite_focus=true

Confirmed in AWS that the service did indeed redeploy with the new task definition and reached a steady state :tada: 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Note: the current configuration of this action doesn't really make much sense because it's deploying to the same place on releases, pushes to main, and workflow dispatches. Once the ECS infrastructure is finalized and we have two environments for it (staging and production) then the pushes to main will only deploy staging and releases (and workflow dispatches) will deploy to the production service. The intention here is to mirror our current production deployment cadence where we deploy to production through a "release" cycle rather than on every push to main. The workflow dispatch is included as a way to rollback to a previous version.

In the future (like maybe by the end of this year/early next year) I'd like to get us to a point where every push to main implies a production deployment through some kind of pattern that looks like this:

1. Merge to main
2. Staging deploys
3. Verify changes in staging
4. Press a button somewhere that triggers the production deployment (maybe a Workflow to dispatch)

We'll need to think about safe guards so someone doesn't accidentally merge to main and then ignore the rest of the process (because then it will fall on subsequent deployers to test the previous merge in staging before they're able to push their new revision to production). We'll also want to try doing things like running e2e tests against the actual deployed staging environment and block the push to production until the e2e tests pass.

In short: there's a _lot_ of work for us to get to the quick "pushing to main implies a production deployment" setting, but we can get there eventually. The work I'm doing now sets us up for that.

cc @WordPress/openverse-maintainers please check out this comment as I believe it's good context for how I'm approaching this work currently and what the goals are right now vs the far future goals we've discussed to varying levels of depth.

**Comment by obulat** [comment]
> get us to a point where every push to main implies a production deployment 

This sounds really dangerous, even if there's a button that you need to press after checking that staging is fine. I wonder if the _checking that staging is fine_ after every PR merge instead of _checking for release every [x] days_ would add more work... Or maybe we would need to change our perceptions of what deploy means :)

**Comment by sarayourfriend** [comment]
> This sounds really dangerous, even if there's a button that you need to press after checking that staging is fine. I wonder if the checking that staging is fine after every PR merge instead of checking for release every [x] days would add more work... Or maybe we would need to change our perceptions of what deploy means :)

There are lots of big projects that follow this pattern (wp-calypso for example as well as other projects I've worked on in the past) and it's pretty safe. There are a bunch of details I'm leaving out like:

1. Rollbacks have to be easy and fast to do
2. Deploying to Production from Staging needs to _block_ on anything that doesn't pass (like e2e tests)
3. Complementing that second point, we need to have robust e2e tests that confirm the entire core user flows are healthy (implementing #1244 which is _well_ past it's due date, I've just realized, is a good step in that direction)
4. We have to trust contributors to do their due diligence when testing their changes in staging so that it doesn't break production

It also helps to have the perspective that if we're deploying every commit to main to staging and then to production after staging is verified, then our production deployment units are _much_ smaller than they are today. You're not worried about "am I going to break the entire app" because 99% of the time the changes you merged only change a small piece of functionality. Even in long-running feature development, we'll leverage feature flags to avoid having to do massive significant deployments.

**Review by zackkrida — APPROVED** [review]
lgtm!
